cobra/core/DictList.py

```diff
@@ -54,7 +54,7 @@ def list_attr(self, attribute):
         """
         return [getattr(i, attribute) for i in self]
 
-    def query(self, search_function, attribute="id"):
+    def query(self, search_function, attribute=None):
```

Comment:
Yeah I was wondering that -- I'll think about it a bit more. Maybe if its a string, you search ID, if its a function, you use the object..

My main concern is that it seems odd to have to pass `None` as an explicit argument.

Not that it really matters, but I think the more generalizable way to get boundary reactions would be 
```python
>>> model.reactions.query(lambda x: x.boundary)
```

That's how I try to do it in the cobrapy methods (i.e., `summary`).